include/proxy/v4/proxy.h

```diff
@@ -0,0 +1,2381 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#ifndef _MSFT_PROXY4_
+#define _MSFT_PROXY4_
+
+#include "proxy_macros.h"
+
+#include <cassert>
+#include <cstddef>
+#include <cstdlib>
+#include <bit>
+#include <concepts>
+#include <exception>
+#include <initializer_list>
+#include <limits>
+#include <memory>
+#include <new>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+#if __STDC_HOSTED__
+#include <atomic>
+#if __has_include(<format>)
+#include <format>
+#endif  // __has_include(<format>)
+#endif  // __STDC_HOSTED__
+
+#if __cpp_rtti >= 199711L
+#include <optional>
+#include <typeinfo>
+#endif  // __cpp_rtti >= 199711L
+
+#if __has_cpp_attribute(msvc::no_unique_address)
+#define ___PRO4_NO_UNIQUE_ADDRESS_ATTRIBUTE msvc::no_unique_address
```

Comment:
Since this macro will be undefined at the end of the file, suggest reverting the name

Reply:
I think every versioned name should carry a clear motivation. For internal macros, versioning doesn’t seem necessary.